server/src/controllers/orderController.ts

```diff
@@ -330,30 +331,32 @@ orderController.put('/refund/:id', async (req, res) => {
     if (!order) {
       return res.status(400).json({error: `Order ${orderID} does not exist`});
     }
-    if (!order.payment_intent) {
+    if (!order.payment_intent || order.order_status !== state.completed) {
       return res.status(400).json({error: `Order ${orderID} is still processing`});
     }
     if (!order.donation && !order.orderticketitems.length && !order.subscriptions.length) {
       return res.status(400).json({error: `Order ${orderID} has already been fully refunded`});
     }
 
     let refundIntent;
-    if (order.payment_intent.includes('comp')) refundIntent = `refund-comp-${order.orderid}`;
-    else if (order.payment_intent.includes('seeded-order')) refundIntent = `refund-seeded-order-${order.orderid}`;
-    else {
+    // eslint-disable-next-line camelcase
+    const {payment_intent} = order;
+    if (payment_intent.includes('comp') || payment_intent.includes('seeded-order')) {
```

Comment:
For sake of not writing code specifically for our test data, does it make sense to just seed the orders with `comp` payment_intents, so they'd just be refunded like comped orders.